docs/source/layout.rst

```diff
@@ -0,0 +1,600 @@
+      FROM tbl_a
+
+#. *No whitespace and a newline is not allowed.* This option is
+   configured using the :code:`inline` setting. The most common example
+   of this is spacing within the parts of qualified identifier e.g.
+   :code:`my_schema.my_table`. If a newline were present between the
+   :code:`.` and either :code:`my_schema` or :code:`my_table`, then
+   the expression would not parse and so no newlines should be allowed.
+
+
+.. _alignedelements:
+
+Aligned elements
+^^^^^^^^^^^^^^^^
+
+A special case of spacing is where elements are set to be aligned
+within some limits. This is not enabled by default, but can be
+be configured to achieve layouts like:
+
+.. code-block:: sql
+
+   SELECT
+      a           AS first_column,
+      b           AS second_column,
+      (a + b) / 2 AS third_column
+   FROM foo AS bar
+
+In this example, the alias expressions are all aligned with each other.
+Practically to configure this, SQLFluff needs to know what elements to
+align and how far to search to find elements which should be aligned
+with eachother. The configuration to achieve this layout is:
+
+.. code-block:: ini
+
+   [sqlfluff:layout:type:alias_expression]
+   # We want non-default spacing _before_ the alias expressions.
+   spacing_before = align
+   # We want to align them within the next outer select clause.
+   # This means for example that alias expressions within the FROM
+   # or JOIN clause would _not_ be aligned with them.
+   align_within = select_clause
+   # The point at which to stop searching outward for siblings, which
+   # in this example would likely be the boundary of a CTE. Stopping
+   # when we hit brackets is usually a good rule of thumb for this
+   # configuration.
+   align_boundary = bracketed
+
+
+Line Breaks
+-----------
+
+When controlling line breaks we are trying to achieve a few different things:
+
+#. Do we have *enough* line breaks that *line length* doesn't become
+   excessive. Long lines are hard to read, especially given that readers
+   may be on varying screen sizes or have multiple windows open. This is
+   (of course) configurable, but the default is 80 characters (in line with
+   the `dbt Labs SQL style guide`_.)
+
+#. Are the positioning of *blank lines* (i.e. lines with nothing other
+   than whitespace on them) appropriate. There are some circumstances
+   where a blank line is *desired* (e.g. between CTEs). There are others
+   where they are not, in particular *multiple blank lines*, for example
+   at the beginning of a file.
+
+#. Where we do have line breaks, are they positioned appropriately and
+   consistently with regards to other elements around them. This is most
+   common when it comes to *commas*, and whether they should be *leading*
+   (e.g. :code:`, my_column`) or *trailing* (e.g. :code:`my_column,`). In
+   less common cases it may also be desirable for some elements to have both
+   and line break *before and after* (e.g. a set operator such as `UNION`).
```

Comment:
The words "both and" read a bit strange to me. I think it's technically correct, but it "surprises" me when I read it.

Could we say something like:
"to have both -- i.e. a line break before and after..."